src/puppetlabs/puppetdb/command.clj

```diff
@@ -345,6 +345,25 @@
     "Returns a core.async publisher to which commands are written after they
      have been processed. The topic for each message is its uuid." ))
 
+(defn make-cmd-processed-message [cmd ex]
+  (merge {:command cmd
+          :id (-> cmd :annotations :id)}
+         (if ex
+           {:exception ex}
+           {})))
```

Comment:
Usually we do this with just `(merge ... (when ex {:exception ex}))` since merge handles nil, same with `conj`.